Mathlib/Topology/ContinuousFunction/CompactlySupported.lean

```diff
@@ -0,0 +1,551 @@
+import Mathlib.Topology.Support
+
+/-!
+# Compactly supported continuous functions
+
+In this file, we define the type `C_c(α, β)` of compactly supported continuous functions and the
+class `CompactlySupportedContinuousMapClass`, and prove basic properties.
+
+## Main definitions and results
+
+This file contains various instances such as `Add`, `Mul`, `SMul C(α, β) C_c(α, β)`.
+When `β` has more structures, `C_c(α, β)` inherits such structures as `AddCommGroup`,
+`NonUnitalRing`, `NormedSpace` and `CstarRing`.
+
+When the domain `α` is compact, `ContinuousMap.liftCompactlySupported` gives the identification
+`C(α, β) ≃ C_c(α, β)`. When the codomain is a metric space, every continuous compactly supported map
+is a bounded continuous function by `bounded`.
+
+## TODO
+
+* Create more intances of algebraic structures (e.g., `NonUnitalSemiring`) once the necessary
+  type classes (e.g., `TopologicalRing`) are sufficiently generalized.
+-/
+
+variable {F α β γ : Type*} [TopologicalSpace α]
+
+open BoundedContinuousFunction Topology Bornology
+
+open Filter Metric
+
+/-- `C_c(α, β)` is the type of continuous functions `α → β` with compact support from a topological
+space to a topological space with a zero element.
+
+When possible, instead of parametrizing results over `f : C_c(α, β)`,
+you should parametrize over `{F : Type*} [CompactlySupportedContinuousMapClass F α β] (f : F)`.
+
+When you extend this structure, make sure to extend `CompactlySupportedContinuousMapClass`. -/
+structure CompactlySupportedContinuousMap (α β : Type*) [TopologicalSpace α] [Zero β]
+    [TopologicalSpace β] extends ContinuousMap α β where
+  /-- The function has compact support . -/
+  hasCompactSupport' : HasCompactSupport toFun
+
+@[inherit_doc]
+scoped[CompactlySupported] notation (priority := 2000)
+  "C_c(" α ", " β ")" => CompactlySupportedContinuousMap α β
+
+@[inherit_doc]
+scoped[CompactlySupported] notation α " →C_c " β => CompactlySupportedContinuousMap α β
+
+open CompactlySupported
+
+section
+
+/-- `CompactlySupportedContinuousMapClass F α β` states that `F` is a type of continuous maps with
+compact support.
+
+You should also extend this typeclass when you extend `CompactlySupportedContinuousMap`. -/
+class CompactlySupportedContinuousMapClass (F : Type*) (α β : outParam <| Type*)
+    [TopologicalSpace α] [Zero β] [TopologicalSpace β] [FunLike F α β]
+    extends ContinuousMapClass F α β : Prop where
+  /-- Each member of the class has compact support. -/
+  hasCompactSupport (f : F) : HasCompactSupport f
+
+end
+
+export CompactlySupportedContinuousMapClass (hasCompactSupport)
+
+namespace CompactlySupportedContinuousMap
+
+section Basics
+
+variable [TopologicalSpace β] [Zero β] [FunLike F α β] [CompactlySupportedContinuousMapClass F α β]
+
+instance : FunLike C_c(α, β) α β where
+  coe f := f.toFun
+  coe_injective' f g h := by
+    obtain ⟨⟨_, _⟩, _⟩ := f
+    obtain ⟨⟨_, _⟩, _⟩ := g
+    congr
+
```

Comment:
Can you add here the lemma
```
protected lemma hasCompactSupport (f : C_c(α, β)) : HasCompactSupport f := f.hasCompactSupport'
```
to make sure that `f.hasCompactSupport` works whenever `f : C_c(α, β)`?

Reply:
yes, done.